tests/components/tesla_wall_connector/test_config_flow.py

```diff
@@ -0,0 +1,207 @@
+    assert result2["title"] == "Tesla Wall Connector"
+    assert result2["data"] == {CONF_HOST: "1.1.1.1"}
+    assert len(mock_setup_entry.mock_calls) == 1
+
+
+async def test_form_cannot_connect(hass: HomeAssistant) -> None:
+    """Test we handle cannot connect error."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    with patch(
+        "tesla_wall_connector.WallConnector.async_get_version",
+        side_effect=WallConnectorConnectionError,
+    ):
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {CONF_HOST: "1.1.1.1"},
+        )
+
+    assert result2["type"] == RESULT_TYPE_FORM
+    assert result2["errors"] == {"base": "cannot_connect"}
+
+
+async def test_form_other_error(
+    mock_wall_connector_version, hass: HomeAssistant
+) -> None:
+    """Test we handle any other error."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    with patch(
+        "tesla_wall_connector.WallConnector.async_get_version",
+        side_effect=Exception,
+    ):
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {CONF_HOST: "1.1.1.1"},
+        )
+
+    assert result2["type"] == RESULT_TYPE_FORM
+    assert result2["errors"] == {"base": "unknown"}
+
+
+async def test_form_already_configured(mock_wall_connector_version, hass):
+    """Test we get already configured."""
+
+    entry = MockConfigEntry(
+        domain=DOMAIN, unique_id="abc123", data={CONF_HOST: "0.0.0.0"}
+    )
+    entry.add_to_hass(hass)
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    with patch(
+        "homeassistant.components.tesla_wall_connector.async_setup_entry",
+        return_value=True,
+    ):
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {CONF_HOST: "1.1.1.1"},
+        )
+        await hass.async_block_till_done()
+
+        assert result2["type"] == "abort"
+        assert result2["reason"] == "already_configured"
+
+    # Test config entry got updated with latest IP
+    assert entry.data[CONF_HOST] == "1.1.1.1"
+
+
+async def test_dhcp_can_finish(mock_wall_connector_version, hass):
+    """Test DHCP discovery flow can finish right away."""
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": config_entries.SOURCE_DHCP},
+        data={
+            HOSTNAME: "teslawallconnector_abc",
+            IP_ADDRESS: "1.2.3.4",
+            MAC_ADDRESS: "DC:44:27:12:12",
+        },
+    )
+    await hass.async_block_till_done()
+    assert result["type"] == "form"
+    assert result["step_id"] == "user"
+
+    result = await hass.config_entries.flow.async_configure(
```

Comment:
We should patch the integration setup function on all create entry results to avoid setting up the whole integration.